feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts

```diff
@@ -2,18 +2,36 @@ import { ChangeDetectionStrategy } from '@angular/core';
 import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
 import { ReactiveFormsModule } from '@angular/forms';
 import { NgSelectModule } from '@ng-select/ng-select';
+import { CommonConfiguratorTestUtilsService } from '@spartacus/product-configurator/common';
 import { Configurator } from '../../../../core/model/configurator.model';
 import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
 import { ConfiguratorAttributeDropDownComponent } from './configurator-attribute-drop-down.component';
 
+function createValue(code: string, name: string, isSelected: boolean) {
+  const value: Configurator.Value = {
+    valueCode: code,
+    name: name,
+    selected: isSelected,
+  };
+  return value;
+}
+
 describe('ConfigAttributeDropDownComponent', () => {
   let component: ConfiguratorAttributeDropDownComponent;
   let fixture: ComponentFixture<ConfiguratorAttributeDropDownComponent>;
+  let htmlElem: HTMLElement;
+
   const ownerKey = 'theOwnerKey';
   const name = 'theName';
   const groupId = 'theGroupId';
   const selectedValue = 'selectedValue';
 
+  const value1 = createValue('1', 'val1', true);
+  const value2 = createValue('2', 'val2', false);
+  const value3 = createValue('3', 'val3', false);
+
+  const values: Configurator.Value[] = [value1, value2, value3];
+
```

Comment:
for this `providers: [ConfiguratorAttributeBaseComponent]`, we want to mock `ConfiguratorAttributeQuantityService`, right? which is the dependent service.

Reply:
@WeizhengSap, yes, we have to mock `ConfiguratorAttributeQuantityService` but we could do it as soon as the delivered version is deprecated. There is a task #11681 not to forget to take care about it. The dependency to `ConfiguratorAttributeQuantityService` comes in the newest version. There are 2 tests: `should not allow quantity when service is missing` & `should not allow quantity actions when service is missing` that check what happens when the `quantityService` is undefined.